adapters/interlocked_win32.c

```diff
@@ -1,4 +1,318 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-#include "azure_c_util/interlocked.h"
\ No newline at end of file
+#include "azure_c_util/interlocked.h"
+#include "windows.h"
+
+IMPLEMENT_MOCKABLE_FUNCTION(, int32_t, interlocked_add, volatile int32_t*, addend, int32_t, value) 
+{
+	/*Codes_SRS_INTERLOCKED_43_001 [ interlocked_add shall atomically add *addend with value and store the result in *addend.]*/
```

Comment:
>	 [](start = 0, length = 1)

Not a big deal, but can we use spaces for indentation? Makes it consistent across the team. But again, not that big of a deal.